### PR TITLE
CASH-103: Lend Increment Button

### PR DESCRIPTION
**Pull request by ksekhri**
* Tracking will be added in the next PR

```
uiexp.lend_increment_button
string
{"name": "LendIncrementButton","enabled": true,"startTime": "2018-08-29T00:00","endTime": "2020-10-14T00:00","control": {"key": "variant-a","name": "Control (variant-a)"},"variants": {"variant-b": {"name": "Lend Increment Button Enabled"}},"distribution": {"variant-a": 0,"variant-b": 1}}
Lend Increment Button Experiment
```

### CODE REVIEW

**Comment by ksekhri** [comment]
@mcstover @emuvente Updated as per our discussion

File: src/components/LoanCards/Buttons/LendButton.vue [hunk 1]
```diff
@@ -69,9 +72,24 @@ export default {
...
+		},
+		trackCash103() {
+			const lendIncrementExperimentVersion = this.apollo.readQuery({
+				query: experimentQuery,
+				variables: { id: 'lend_increment_button' },
+			});
+			// eslint-disable-next-line max-len
+			const lendIncrementExperimentVersionString = _get(lendIncrementExperimentVersion, 'experiment.version') || null;
+			if (lendIncrementExperimentVersionString === 'variant-b') {
+				this.$kvTrackEvent('lending', 'EXP-CASH-103-Jan2019-click-lendnow', this.price);
```
Comment (mcstover) [inline src/components/LoanCards/Buttons/LendButton.vue]: Please make 'lending', 'Lending' to match the category already in use.

File: src/pages/Lend/LendByCategoryPage.vue [hunk 2]
```diff
@@ -321,6 +323,18 @@ export default {
...
+		// Read assigned version of lend increment button experiment
+		const lendIncrementExperimentVersion = this.apollo.readQuery({
+			query: experimentQuery,
+			variables: { id: 'lend_increment_button' },
+		});
+		const lendIncrementExperimentVersionString = _get(lendIncrementExperimentVersion, 'experiment.version') || null;
+		if (lendIncrementExperimentVersionString === 'variant-a') {
+			this.$kvTrackEvent('lending', 'EXP-CASH-103-Jan2019', 'a');
+		} else if (lendIncrementExperimentVersionString === 'variant-b') {
+			this.$kvTrackEvent('lending', 'EXP-CASH-103-Jan2019', 'b');
```
Comment (mcstover) [inline src/pages/Lend/LendByCategoryPage.vue]: Same with these 2: Lending

File: src/components/LoanCards/Buttons/LendButton.vue (hunk 1 above)
Comment (ksekhri) [reply src/components/LoanCards/Buttons/LendButton.vue]: Nice catch :+1:

File: src/pages/Lend/LendByCategoryPage.vue (hunk 2 above)
Comment (ksekhri) [reply src/pages/Lend/LendByCategoryPage.vue]: Done :+1:

**Comment by ksekhri** [comment]
@mcstover that's an interesting question. I ran across this in this PR itself and created a `trackCash103` method inside `LendButton.vue`. The thought here was we got the experiment data from the settings manager and processed it through our local resolver in `LendByCategoryPage.vue`. Then in `LendButton.vue` I read the query and decided what tracking to do inside `trackCash103`. In the future we can make that method more general to track other experiments as well.

**Pull request merged**